stl/inc/typeindex

```diff
@@ -40,9 +41,12 @@ public:
 
 #if _HAS_CXX20
     _NODISCARD strong_ordering operator<=>(const type_index& _Right) const noexcept {
-        return *_Tptr == *_Right._Tptr      ? strong_ordering::equal
-             : _Tptr->before(*_Right._Tptr) ? strong_ordering::less
-                                            : strong_ordering::greater;
+        // TRANSITION, DevCom-10326599, should rely on a stable interface
+        if (_Tptr == _Right._Tptr) {
+            return strong_ordering::equal;
+        }
+
+        return _CSTD strcmp(_Tptr->raw_name() + 1, _Right._Tptr->raw_name() + 1) <=> 0;

```

Comment:
No change requested, note to final reviewer: This `raw_name() + 1` looks like 5 squirrels out of 5 :chipmunk: but it's exactly the correct technique.

* `__std_type_info_compare()` in `std_type_info.cpp` ([internal link][std_type_info.cpp]) compares `strcmp(lhs->_DecoratedName + 1, rhs->_DecoratedName + 1)`.
* `type_info::raw_name()` in `vcruntime_typeinfo.h` ([internal link][vcruntime_typeinfo.h]) says `return _Data._DecoratedName;`.
* The character at index 0 appears to be a dot; I'm not sure why that's stored, or why `raw_name()` exposes this, but the implementation here is correct.

[std_type_info.cpp]: https://devdiv.visualstudio.com/DevDiv/_git/msvc?path=/src/vctools/crt/vcruntime/src/eh/std_type_info.cpp&version=GBprod/fe&_a=contents
[vcruntime_typeinfo.h]: https://devdiv.visualstudio.com/DevDiv/_git/msvc?path=%2Fsrc%2Fvctools%2Fcrt%2Fvcruntime%2Finc%2Fvcruntime_typeinfo.h&version=GBprod%2Ffe&_a=contents